### PR TITLE
Add compiler flags

### LINKED ISSUE #435: Add extra compilation options for clio
Current set of options and warnings when building `clio` is quite narrow (see `CMake/settings.cmake`).
Let's implement more safety nets. 

Add at least these recommended/standard options: 
```
-Wextra 
-Wshadow
-Wnon-virtual-dtor
-pedantic
```

This will break compilation of `clio` which will need to be fixed for all supported compilers.

### PR DESCRIPTION
**Pull request by kuznetsss**
Fixes #435

### CODE REVIEW

File: src/etl/ETLService.h [hunk 1]
```diff
@@ -230,6 +230,15 @@ class ETLService
     void
     monitor();
 
+    /**
+     * @brief Monitor the network for newly validated ledgers and publish them to the ledgers stream
```
Comment (godexsoft) [inline src/etl/ETLService.h:234]: Is this description correct?

File: src/etl/LoadBalancer.cpp [hunk 2]
```diff
@@ -147,7 +147,7 @@ LoadBalancer::forwardToRippled(
     std::string const& clientIp,
     boost::asio::yield_context yield) const
 {
-    srand((unsigned)time(0));
+    srand(static_cast<unsigned>(time(0)));
```
Comment (godexsoft) [inline src/etl/LoadBalancer.cpp:150]: Oof. Let's rewrite srand/rand to normal c++ random generation at some point 🔢

File: unittests/data/cassandra/BackendTests.cpp [hunk 3]
```diff
@@ -655,19 +664,20 @@ TEST_F(BackendCassandraTest, Basic)
...
                 } while (cursor);
                 EXPECT_EQ(retData.size(), data.size());
                 for (size_t i = 0; i < retData.size(); ++i)
                 {
-                    auto [txn, meta, seq, date] = retData[i];
-                    auto [hash, expTxn, expMeta] = data[i];
-                    EXPECT_STREQ((const char*)txn.data(), (const char*)expTxn.data());
-                    EXPECT_STREQ((const char*)meta.data(), (const char*)expMeta.data());
+                    auto [txn, meta, _, __] = retData[i];
+                    auto [___, expTxn, expMeta] = data[i];
```
Comment (godexsoft) [inline unittests/data/cassandra/BackendTests.cpp:678]: This looks questionable 😆 I don't really have a better option tho.

**Review by godexsoft — DISMISSED** [review]
LGTM! 🚀

**Review by cindyyan317 — APPROVED** [review]

**Pull request merged**